nest-forms-backend/src/nases/utils-services/tokens.nases.service.ts

```diff
@@ -409,48 +431,79 @@ export default class NasesUtilsService {
     let subject: string = form.id
     let mimeType = 'application/x-eform-xml'
     let encoding = 'XML'
-    let attachments = ''
+    let attachments: XmlObjectList = []
 
     if (isSlovenskoSkTaxFormDefinition(formDefinition)) {
       subject = 'Podávanie daňového priznanie k dani z nehnuteľností' // TODO fix in formDefinition, quickfix here formDefinition.messageSubjectDefault
       mimeType = 'application/vnd.etsi.asic-e+zip'
       encoding = 'Base64'
       attachments = await this.createAttachmentsIfExists(form, formDefinition)
     }
-    envelope = `
-            <?xml version="1.0" encoding="utf-8"?>
-            <SKTalkMessage xmlns="http://gov.sk/SKTalkMessage" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:xsd="http://www.w3.org/2001/XMLSchema">
-              <EnvelopeVersion>3.0</EnvelopeVersion>
-              <Header>
-                <MessageInfo>
-                  <Class>EGOV_APPLICATION</Class>
-                  <PospID>${pospID}</PospID>
-                  <PospVersion>${pospVersion}</PospVersion>
-                  <MessageID>${form.id}</MessageID>
-                  <CorrelationID>${correlationId}</CorrelationID>
-                </MessageInfo>
-              </Header>
-              <Body>
-                <MessageContainer xmlns="http://schemas.gov.sk/core/MessageContainer/1.0">
-                  <MessageId>${form.id}</MessageId>
-                  <SenderId>${senderId}</SenderId>
-                  <RecipientId>${
-                    process.env.NASES_RECIPIENT_URI ?? ''
-                  }</RecipientId>
-                  <MessageType>${pospID}</MessageType>
-                  <MessageSubject>${subject}</MessageSubject>
-                  <Object Id="${form.id}" IsSigned="${
-                    isSigned ? 'true' : 'false'
-                  }" Name="${title}" Description="" Class="FORM" MimeType="${mimeType}" Encoding="${encoding}">${message}</Object>
-    ${attachments}
-                </MessageContainer>
-              </Body>
-            </SKTalkMessage>
-      `
+
+    attachments.push({
+      $: {
+        Id: form.id,
+        IsSigned: isSigned ? 'true' : 'false',
+        Name: title,
+        Description: '',
+        Class: 'FORM',
+        MimeType: mimeType,
+        Encoding: encoding,
+      },
+      // TODO When message should be created as object
+      _: message,
```

Comment:
We can use xml2js to parse it to object and embed it here, if parsing fails we can throw an error (this would also check if message is correct XML - which we don't do now).

Reply:
Just as a note: it will only check for valid XML and will let any potential xml injection pass. (this should be ok at this stage)